src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -2084,8 +2091,14 @@ bool isRefEscape
                                         _compilation.IsPeVerifyCompatEnabled,
                                         stackLocalsOpt: null))
                     {
+#if DEBUG
+                        CheckVisited(receiver);
+#endif
                         // Equivalent to a non-ref local with the underlying receiver as an initializer provided at declaration 
                         receiver = new BoundCapturedReceiverPlaceholder(receiver.Syntax, receiver, _localScopeDepth, receiver.Type).MakeCompilerGenerated();
+#if DEBUG
+                        MarkVisited(receiver);
```

Comment:
Consider removing this call if we end up having `MarkVisited` be a no-op on placeholders.